test/markup/handlebars/block-expression-variants-in-param.txt

```diff
@@ -1,7 +1,5 @@
-text {{#abc "lite]'ral}}segment" }}a{{/abc}}
+Block expression variants with literal segments in helper param:
```

Comment:
I was hoping you would justify (perhaps I'm not getting the importance) or remove these to make it easier to review the actual meat of this commit and not just all the dressing...

Reply:
Both titles and dots are there because I used `developer.html` to inspect the test results.

At first I started to insert one file after another into `developer.html`. Then I started concatenating files. The titles in the files make it easier to see, which file the tests actually belong to (as you can see in the screenshots above)

And sometimes, the output looked good, but on a second glance, I could see that a mode was spanning into the next textcase. The dots `...` are there to ensure that the mode ends where it should. The dots never be highlighted but always be displayed as "content" or "xml".

I think the dots are important. I can remove the titles, although if I start working on the language again, for whatever reason, I would like to have them in there.

Reply:
> And sometimes, the output looked good, but on a second glance, I could see that a mode was spanning into the next textcase. The dots ... are there to ensure that the mode ends where it should. The dots never be highlighted but always be displayed as "content" or "xml".

This sounds like a real problem, but one that needs a better solution than adding a ton of ...s everywhere.  Have you have any lightbulb moment while working on tests on perhaps a better way of doing this?

Reply:
I'm not completely sure I"m following how you're using developer.html... but it sounds like perhaps what your'e doing could/shoudl be automated rather than adding ... and titles everywhere you just run `test php in devtool` and it loads up the devtool with ALL the tests queued with separators, etc...

You can do that without adding all those things to the individual files.

Reply:
I think the dots are a valid method to ensure the transition from tag to text. Handlebars is primarily text, with some tags inserted, so we should put some text sections between tags in the test cases. The dots are the most unobtrusive chars I can imagine.

Reply:
As for the headers. I didn't want to implement something new. I just used used `ls`, `grep`, `xargs` and `cat` to generate a file with all test cases and copied that into the developer.html

I agree that a solution to view all test cases would be better, but currently there is none.

Reply:
I can remove the headers, but I think there should still be some text around the tags

Reply:
> I think the dots are a valid method to ensure the transition from tag to text. Handlebars is primarily text, with some tags inserted, so we should put some text sections between tags in the test cases. 

You might have to try harder to explain the purpose of this to me again, but I still not sure I will be persuaded.  It feels like (for some reason) you're wanting to a way to SEPARATE individual test cases within a file.  That isn't a problem unique to HTMLBars... and I already dislike this solution and I'd hate to see it find it's way into all our other test cases (plus one can imagine a grammar where ... has real meaning, then UGH.  

And also dislike the idea that every grammar will just invent their own choice way of separating test cases...

There may indeed be a real problem here, but I don't think this is the right solution.

---

It might not be completely terrible if it was already there, but when you try to change the test AND add the ... at the same time the changes become impossible to review because the ... generates a ridiculous amount of noise in the diffs.

---

Lets start higher level.... can you explain clearly the problem you're trying to solve with "..." and why led you to add them in the first place?  I'm not sure I follow that part.

Reply:
Currently in other cases we accomplish this with blank lines, or perhaps a comment (which then also gets highlighted, which adds noise)... I'd be open to coming up with a smarter encoding format for these test cases, but it needs to be clear, simple and something we can do globally in a single pass, not something that interferes with PRs and get added piecemeal.

Reply:
>  Handlebars is primarily text, with some tags inserted, so we should put some text sections between tags in the test cases.

I think I may disagree with your premise here, at least in the general case.  I think of markup tests (at least THIS many and this specific) as unit tests.  They should be as specific as possible and test as narrowly as possible.  

Testing that the "overall" grammar works might be a markup test, but that could be a SINGLE test showing that tags and text mix well, it doesn't need to be proven over and over in a every single unit test.

Reply:
Take another grammar for example that formats numbers inside of text... with 100 varieties of numbers... A good test might be:

```
# numbers mixed with content
This is a line with 32 numbers.
How do pigs fly if they weight 1,000 pounds?
# 100 different tests for individual number formats...
12323
123,320
123,233.232
0xFFFF
...
```
Adding  words to every line is noise and gets in the way of what the tests are trying to prove (that the number recognition regex works).